packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -762,11 +811,29 @@ export class GarbageCollector implements IGarbageCollector {
         // Basically, it was written in the current GC version.
         if (result.wasSummaryTracked) {
             this.latestSummaryGCVersion = this.currentGCVersion;
+            if (this.trackGCBlobState) {
+                this.latestSerializedSummaryState = this.pendingSerializedSummaryState;
```

Comment:
What happens if pending state is undefined? Should we assert that it should not be undefined?

Reply:
No, then we rewrite the whole blob

Reply:
Maybe another to put it - If the summary is tracked (result.wasTracked) and GC is running, pending state should always be available. We can say that because this is the client that submitted the summary for which we got this ack.

Reply:
The first time a summary is submitted, pending state is not always available in the case `WriteDataAtRoot === false`